project/src/main/java/com/google/sps/servlets/UserProfileUpdateServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.User;
+import com.google.sps.utilities.CommonUtils;
+import com.google.sps.utilities.DatabaseConstants;
+import com.google.sps.utilities.DatabaseWrapper;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import javax.json.Json;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles user profile edits. */
+@WebServlet("/profile-update")
+public class UserProfileUpdateServlet extends HttpServlet {
+  private static final String email =
+      UserServiceFactory.getUserService().getCurrentUser().getEmail();
+  private DatabaseWrapper databaseWrapper =
+      new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
+
+  /** Writes out information for the user corresponding to the logged-in email */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<User> userOptional = databaseWrapper.readUserFromEmail(email);
+    if (!userOptional.isPresent()) {
+      String userJson =
+          Json.createObjectBuilder()
+              .add("name", "anonymous")
+              .add("email", email)
+              .add("interests", CommonUtils.createJsonArray(new HashSet<>()))
+              .add("skills", CommonUtils.createJsonArray(new HashSet<>()))
+              .build()
+              .toString();
+    }
+    User user = userOptional.get();
+    String userJson =
```

Comment:
Else?

Reply:
BTW, also leave a TODO to add test for this.

Reply:
Ah, missed the else - done.